internal/controller/components/reconciler.go

```diff
@@ -156,6 +185,44 @@ func Reconcile(
 			continue
 		}
 
+		if needsPullSecret {
+			providerSecret, skipCreation, err := makeProviderSecret(registryUsername, registryPassword, component.name, opts.Namespace, template)
+			if err != nil {
+				errMsg := fmt.Sprintf("failed to generate provider secret: %s", err)
+				updateComponentsStatus(statusAccumulator, component, template, errMsg)
+				errs = errors.Join(errs, errors.New(errMsg))
+				continue
+			}
+
+			if !skipCreation {
+				l.Info("Creating provider variable secret for " + component.name)
+				if err := mgmtClient.Create(ctx, providerSecret); client.IgnoreAlreadyExists(err) != nil {
```

Comment:
It should be created using the regional client (the secret must exist on the regional cluster to deploy providers on it). In case of the management cluster, `rgnlClient == mgmtClient`.
```suggestion
				if err := rgnlClient.Create(ctx, providerSecret); client.IgnoreAlreadyExists(err) != nil {
```